src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/CustomPipelineAction.java

```diff
@@ -32,56 +40,75 @@ public static void main(String[] args) {
             .addAction(new UpcaseFieldsAction().addField("Country"));
         
         pipeline.run();
+        
+        System.out.println("Generated Code:" + System.lineSeparator());
+        System.out.println(pipeline.getJavaCode().getSource());
     }
     
-}
+    static class UpcaseFieldsAction extends PipelineAction {
 
-class UpcaseFieldsAction extends PipelineAction {
+        private final List<String> fieldNameList;
 
-    private final List<String> fieldNameList;
+        protected UpcaseFieldsAction() {
+            super("transform-upcase-fields", "Upper Case Fields");
+            fieldNameList = new ArrayList<>();
```

Comment:
Move this initialization up as part of the declaration line.

Reply:
Done.